### ISSUE TITLE
DevDojo Auth breaks Jetstream's Teams support

### ISSUE THREAD

**Issue by ch4r-ch4r** [thedevdojo/auth#85]
**Describe the bug**
On a new Jetstream application (with "Teams" support enabled) registering a new User does not automatically create its own Team.

**To Reproduce**
1. Create a new Jetstream app (following [Jetstreams docs](https://jetstream.laravel.com/installation.html))
1.1  Run `composer create-project laravel/laravel example-app`
1.2 Run `cd example-app`
1.3 Run `composer require laravel/jetstream`
1.4 Run `php artisan jetstream:install livewire --teams`

At this point, on the `/register` view, when a new User is registered, a new Team for the registered User is also created in the `teams` table.

2. Install Dev Dojo's Auth package for Jetstream app (following [the docs](https://devdojo.com/auth/docs/install/jetstream/))
2.1 Install the Auth Package
2.2 Publish the Fortify config
2.3 Enable the Name Field for Registration

3. Go through the Register user flow.

**Expected behavior**
After submitting the Register form in `/register`:
* The user is redirected to the `/login` view.
* User can login with their credentials, and land on the Dashboard which Displays their personal Team and User at the top right corner.

**Actual behavior**
After submitting the Register form in `/register`:
* The user receives an Internal Server Error.
```
ErrorException
Attempt to read property "name" on null
```
* This is due to the Team (`teams` table) not being created along with the User, then the `navigation-menu` tries to access the Team's Name and crashes the app.

**Screenshots**
![2024-07-04 at 20 09 37](https://github.com/thedevdojo/auth/assets/78950093/9b35c4e4-8e8b-47bd-93d8-6f2c9ce195f8)

**System Details**
 - OS: MacOS 14.5 (23F79)
 - Browser: Chrome 128.0.6559.0 (Official Build) dev (arm64)
 - Auth Version: "^1.0" (1.0.3)
 - PHP: 8.3

**Comment by arbiyanto** [comment]
I think the current behavior is expected to not have any team. You can simply create the team by adding a new Listener to the `Illuminate\Auth\Events\Registered` Event.

**Comment by ch4r-ch4r** [comment]
Right, if we just do `php artisan jetstream:install livewire` then install DevDojo/Auth and register a new User all is good (there are no Teams involved here).

The part i'm not following is that installing `php artisan jetstream:install livewire --teams` automatically will create a Team for each new User. But if you install DevDojo/Auth then it interferes with the Team creation, only the User will be created.

Does that make sense?

**Comment by kurucu** [comment]
When JetStream is installed with --teams, it uses [an action](https://github.com/laravel/jetstream/blob/5.x/stubs/app/Actions/Fortify/CreateNewUserWithTeams.php) to create each user, which automatically creates a "personal" team for them.

As such, JetStream assumes all users belong to at least one team.

You can either:
- Make it so that DevDojo creates a personal team in a similar way; or,
- Automate/instruct developers to make amendments to files (generated by JetStream stubs) so that they are able to support users with empty `teams()`, and with `currentTeam` being null.

**Comment by kurucu** [comment]
I think it would be good to solve this by adding similar "switch" when installing DevDojo Auth, to enable Teams (or not).

This way, other features can be attached (in the future) to the Teams implementation that aren't useful to non-Teams e.g:
- Customised creation of users (like above)
- Customised authentication per Team
- Inclusion of Middleware that can be used to protect Teams routes, e.g. for Corporate sign on rather than Social sign on.
- etc

And most importantly, keeps non-Teams apps cruft-free.

**Comment by Uhasith** [comment]
I'm also facing this issue. It is better to add this to the installation guide with the Jetstream app. When users register it is always better to create a personal team.
